### PR TITLE
[fix] get_cased_path doesn't raise when path does not exists

### PR DESCRIPTION
**Pull request by jgsogo**
Fix/improve implementation of `get_cased_path`: if you call it with a non-existing path, it will work only on the existing part of the path and leave the rest as is.

### CODE REVIEW

File: conans/test/util/unix_path_test.py [hunk 1]
```diff
@@ -12,16 +12,36 @@
...
+        self.assertEqual(p1, get_cased_path(os.path.join(folder, "myfolder", "subfolder")))
+
+    def test_case_not_existing(self):
+        try:
+            non_existing_path = os.path.abspath(
+                os.path.join("this", "Path", "does", "NOT", "Exists"))
+            p = get_cased_path(non_existing_path)  # If not exists from the root, returns as is
+            self.assertEqual(p, non_existing_path)
+        except Exception as e:
+            self.fail("Unexpected exception: %s" % e)
```
Comment (memsharded) [inline conans/test/util/unix_path_test.py]: An uncontrolled exception will make the test fail anyway, don't use try-except

File: conans/test/util/unix_path_test.py [hunk 2]
```diff
@@ -12,16 +12,36 @@
...
+            folder = get_cased_path(temp_folder())
+            p1 = os.path.join(folder, "MyFolder", "Subfolder")
+            mkdir(p1)
+
+            non_existing_path = os.path.join(folder, "myfolder", "subfolder", "not-existing")
+            # The first part of the path will be properly cased.
+            self.assertEqual(os.path.join(p1, "not-existing"),
+                             get_cased_path(non_existing_path))
+        except Exception as e:
+            self.fail("Unexpected exception: %s" % e)
```
Comment (memsharded) [inline conans/test/util/unix_path_test.py]: Same as above

File: conans/test/util/unix_path_test.py (hunk 1 above)
Comment (jgsogo) [reply conans/test/util/unix_path_test.py]: I prefer having it explicit, but I can remove it... less lines of code ^^

Comment (memsharded) [reply conans/test/util/unix_path_test.py]: Yes, much better, thanks!

**Pull request merged**